js/src/PanZoomModel.ts

```diff
@@ -31,41 +31,41 @@ export const PanZoomModel = basemodel.BaseModel.extend({
             scales: {},
             allow_pan: true,
             allow_zoom: true
-        });
-    },
+        };
+    }
 
-    initialize: function() {
-        PanZoomModel.__super__.initialize.apply(this, arguments);
+    initialize(attributes, options) {
+        super.initialize(attributes, options);
         this.on("change:scales", this.snapshot_scales, this);
         this.snapshot_scales();
-    },
+    }
 
-    reset_scales: function() {
-        var that = this;
-        widgets.resolvePromisesDict(this.get("scales")).then(function(scales: any) {
+    async reset_scales() {
+        widgets.resolvePromisesDict(this.get("scales")).then((scales: any) => {
             _.each(Object.keys(scales), function(k) {
                 _.each(scales[k], function(s: any, i) {
```

Comment:
Maybe you need to make them arrow functions?

Reply:
oops, I did halfway, but then aborted, but you're right that it's wrong now.

Reply:
Done!